future/transaction.go

```diff
@@ -618,6 +621,7 @@ func MakeApplicationCreateTx(
 	accounts []string,
 	foreignApps []uint64,
 	foreignAssets []uint64,
+	appBoxReferences []types.AppBoxReference,
```

Comment:
Unfortunately adding a new function parameter is a breaking change, and a pretty significant one at that. This would cause all downstream code that uses one of these common functions to have a compilation error.

In the past we've had to solve this by introducing a new function, like `MakeApplicationCreateTxWithExtraPages`, which contains the new parameter, and leaving the original function unmodified.

To avoid this annoyance going forward, we really should replace these functions with either ones that accept a single "parameters" struct, or with a builder pattern, but I expect that's a bit out of scope at this time.